fdbserver/RESTKmsConnector.actor.cpp

```diff
@@ -0,0 +1,1143 @@
+/*
+ * RESTKmsConnector.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbserver/RESTKmsConnector.actor.h"
+
+#include "fdbclient/FDBTypes.h"
+#include "fdbclient/rapidjson/document.h"
+#include "fdbclient/rapidjson/rapidjson.h"
+#include "fdbclient/rapidjson/stringbuffer.h"
+#include "fdbclient/rapidjson/writer.h"
+#include "fdbrpc/HTTP.h"
+#include "fdbserver/KmsConnectorInterface.h"
+#include "fdbrpc/RESTClient.h"
+#include "flow/Arena.h"
+#include "flow/EncryptUtils.h"
+#include "flow/Error.h"
+#include "flow/FastRef.h"
+#include "flow/IRandom.h"
+#include "flow/Knobs.h"
+#include "flow/Trace.h"
+#include "flow/UnitTest.h"
+
+#include <cstdio>
+#include <fstream>
+#include <ios>
+#include <memory>
+#include <queue>
+#include <sstream>
+#include <sys/fcntl.h>
```

Comment:
These includes make the windows build fail. I am not yet sure which functionality here is used, but we should try to find a way around Windows problems.

Reply:
For testing I used `mkstemp` for testing "file based validation tokens" scenarios, my suspicion is this causing the failures. 
Do we know how to test windows build locally before updating the PR?
Thanks

Reply:
We don't have a good way testing the Windows build -- I am fully relying on the Windows CI.

For this particular issue, I would probably guard the includes with `#ifdef __unixish__` and then also the test. `mkstemp` is not a C or C++ standard function but a posix extension. Therefore you need to guard it.

Reply:
Should we just use the existing fdb file interfaces to create this file? that way it works in simulation? What is the reason to use mkstemp?